numpy/core/tests/test_dtype.py

```diff
@@ -447,7 +448,8 @@ def test_shape_invalid(self):
 
     def test_alignment(self):
         #Check that subarrays are aligned
-        t1 = np.dtype('1i4', align=True)
+        with pytest.warns(FutureWarning):
+            t1 = np.dtype('1i4', align=True)
         t2 = np.dtype('2i4', align=True)
```

Comment:
Does changing this to (1)i4 fix the warning? We should try to write our tests to not rely on deprecated behavior for anything other than testing the deprecation itself.

Reply:
No, right now `"(1)i4"` is treated as `1i4` (both return a scalar i4 type).  IOW when specifying dtypes as strings there's no way to generate a shape-(1,) field.

Reply:
(1,)i4 is also no good?

Reply:
Ah, (1,)i4 works, edited.